syntax = "proto3";

import "beams/service/remote_calls/generic_message.proto"; // Empty
import "beams/service/remote_calls/heartbeat.proto"; // HeartBeatReply
import "beams/service/remote_calls/command.proto"; // CommandMessage
import "beams/service/remote_calls/behavior_tree.proto";

// Sequencer Service Definition
service BEAMS_rpc {
  // Enqueue a sequence command of varying priority
  rpc enqueue_command (CommandMessage) returns (HeartBeatReply) {}

  // Heart beat message for clients
  rpc request_heartbeat (Empty) returns (HeartBeatReply) {}

  // Detailed message for clients
  rpc request_tree_details (NodeId) returns (TreeDetails) {}
}
